enctests/testframework/main.py

```diff
@@ -386,9 +418,19 @@ def run_tests(args, test_configs, timeline):
             # Run tests and get a dict of resulting media references
             results = encoder.run_wedges()
 
+            testtype = "vmaf"
+            comparisontestinfo = {}
+            if "comparisontest" in test_config:
+                comparisontestinfo = test_config.get("comparisontest", {})
+                testtype = comparisontestinfo.get("testtype", "vmaf")
+
             # Compare results against source
             for test_name, test_ref in results.items():
-                vmaf_compare(source_clip, test_ref, test_name)
+                if testtype != "idiff":
+                    # Run vmaf by default.
+                    vmaf_compare(source_clip, test_ref, test_name)
+                else:
+                    idiff_compare(source_clip, test_ref, test_name, test_config.get("comparisontest", {}))
```

Comment:
Iterate over test types as suggested in the yaml part to not lock into just these two test types.

Reply:
Should be addressed.